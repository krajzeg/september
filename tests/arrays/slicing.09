array := [0,1,2,3,4,5]

print("This simple slice should print '1, 2':", array[1..2])
print("And the length should be 2:", array[1..2].length())

print("Open-ended slice should be the same:", array[1...3])
print("And the length should still be 2:", array[1...3].length())

print("Out of bounds indices should be OK, so this should be 4, 5:", array[4..10])
print("And the length should be truncated to 2:", array[4..10].length())

print("If both indices are out of bounds, the result should be empty:", array[8..23])
print("And the length should be 0:", array[8..23].length())

print("If the 'to' index is less than 'from', the result is empty:", array[5..3])
print("And the length should be 0:", array[5..3].length())

print("array[3..end] should be legal and print 3, 4, 5:", array[3..end])

print("Negative indices should work, so this should be 2, 3, 4:", array[(-4)..(-2)])
print("With a length of 3:", array[(-4)..(-2)].length())

