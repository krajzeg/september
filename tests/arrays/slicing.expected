This simple slice should print '1 2':
1
2
And the length should be 2: 2
Open-ended slice should be the same:
1
2
And the length should still be 2: 2
Out of bounds indices should be OK, so this should be '4 5':
4
5
And the length should be truncated to 2: 2
If both indices are out of bounds, the result should be empty:
And the length should be 0: 0
array[2..end] should print 2,3,4,5:
2
3
4
5
